src/XMakeBuildEngine/Construction/Solution/ProjectInSolution.cs

```diff
@@ -305,11 +305,8 @@ internal bool CanBeMSBuildProjectFile(out string errorMessage)
 
                 if (mainProjectElement != null && mainProjectElement.LocalName == "Project")
                 {
-                    if (String.Compare(mainProjectElement.NamespaceURI, XMakeAttributes.defaultXmlNamespace, StringComparison.OrdinalIgnoreCase) == 0)
-                    {
-                        _canBeMSBuildProjectFile = true;
-                        return _canBeMSBuildProjectFile;
-                    }
+                    _canBeMSBuildProjectFile = true;
```

Comment:
I don't think so. This seems to be a way to "roughly" check if a file is an msbuild file. It's not as accurate as it was before, but at least checks to see if it has a <project> element at the root. I would say keep it. It's also used in quite a few places.

---
In reply to: [80768595](https://github.com/Microsoft/msbuild/pull/1089#discussion_r80768595) [](ancestors = 80768595)